wordpress-mobile/WordPress-iOS
# Recommend App: Implement share button in Me screen

Refs #16995
Resolves #16996
Based on #17007

## Summary
This change adds the share app button to the Me screen when the feature flag is enabled. To add support for activity indicator, the existing `NavigationItemRow` is extended to accept a `loading` Boolean parameter that defaults to `false`. When the value is `true`, the `accessoryView` displays an `UIActivityIndicator` instead of showing an `accessoryType`.

**Note:** In this change, the share app button is added in both the Me and About screen, so both can be verified together in this PR. (cc: @mattmiklic)

Here's a quick video on how this feature looks:

https://user-images.githubusercontent.com/1299411/129264929-c4e838a2-7be1-4e6c-98eb-7caf46e7f61c.mov


## To Test
- Go to App Settings, and turn on the `Recommend App to Others` feature flag.
- Go back to Me screen.
- Verify that the share button is shown.
- Tap the share button, and verify that a loading indicator is shown before the share sheet is presented.
- Dismiss the share sheet.
- Tap the share button again, and verify that the loading indicator is no longer shown; the share sheet should be displayed immediately.

## Regression Notes
1. Potential unintended areas of impact
Some parts of the Me screen is not displayed correctly. May impact self-hosted / non-WP accounts.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manually tested the screen, with and without the feature flag. Also tested with self-hosted accounts.

3. What automated tests I added (or what prevented me from doing so)
n/a

PR submission checklist:
